lib/Utils/Backends.pm

```diff
@@ -288,4 +289,58 @@ sub is_ssh_installation {
           (($videomode =~ /text/) && (is_pvm || is_s390x)));
 }
 
+=head2 update_ssh_console_netconfig
+
+Update the network configuration using current netconfig settings. This will help
+refresh file like /etc/resolv.conf, so associated machine will get the latest or
+corrected name servers to continue to function as normal, for example, establish 
+connection back to worker machine by using FQDN or communicate with system outside 
+using FQDN. If "netconfig update -f" can not help, adding functioning name servers
+into /etc/resolv.conf provided by worker machine. Record failure and even terminate 
+test run if host fails to communicate outside at the last. Arguments are address 
+of the system on which netconfig will operate and the number of retries.
+
+=cut
+
+sub update_ssh_console_netconfig {
+    my (%args) = @_;
+    $args{dest} //= 'localhost';
+    $args{retry} //= 3;
+
+    my $pingaddr = version_utils::is_opensuse() ? 'openqa.opensuse.org' : 'openqa.suse.de';
+    my $sshcmd = $args{dest} eq 'localhost' ? "" : "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root\@$args{dest}";
+    my $pingret = script_run("$sshcmd timeout --kill-after=1 --signal=9 20 ping -c5 $pingaddr");
+    while ($args{retry} > 0 and $pingret != 0) {
+        script_run("$sshcmd timeout --kill-after=1 --signal=9 20 netconfig update -f");
+        $pingret = script_run("$sshcmd timeout --kill-after=1 --signal=9 20 ping -c5 $pingaddr");
+        $args{retry} -= 1;
+    }
+    save_screenshot;
+    if ($pingret != 0) {
+        my $resolv_file = "/etc/resolv.conf";
+        assert_script_run("$sshcmd sed -i -r \'/^nameserver.*\$/d\' $resolv_file") if (script_run("$sshcmd cat $resolv_file | grep \"^nameserver.*\$\"") == 0);
+        my $nameserver = "";
+        open my $fh, '<', $resolv_file or die "Unable to open $resolv_file for input";
+        while (my $line = <$fh>) {
+            if ($line =~ /^nameserver.*$/) {
+                my $temp = $line;
+                $temp =~ s/\n//g;
+                $nameserver .= "\\n$temp";
+            }
+        }
```

Comment:
Can you explain what you want to do here as I don't fully understand it. Regarding reading/writing files in Perl I suggest to use https://docs.mojolicious.org/Mojo/File#slurp but maybe we can further simplify here

Reply:
It performs:
1. open /etc/resolv.conf file on worker machine. Main content of this file is as below:
```
### /etc/resolv.conf is a symlink to /run/netconfig/resolv.conf
### autogenerated by netconfig!
#
# Before you change this file manually, consider to define the
# static DNS configuration using the following variables in the
# /etc/sysconfig/network/config file:
#     NETCONFIG_DNS_STATIC_SEARCHLIST
#     NETCONFIG_DNS_STATIC_SERVERS
#     NETCONFIG_DNS_FORWARDER
# or disable DNS configuration updates via netconfig by setting:
#     NETCONFIG_DNS_POLICY=''
#
# See also the netconfig(8) manual page and other documentation.
#
### Call "netconfig update -f" to force adjusting of /etc/resolv.conf.
search yyyyy zzzzz
nameserver xxxxx
nameserver xxxx

```
2. Iterate over the whole file line by line
3. If a line matches pattern ```^nameserver.*$```, then record its value to ```$temp``` and remove ```\n```
4. Append each matched line to ```$nameserver``` which takes the form of a single string ```\nnameserver\nnameserver2\nnameserver3```
5. ```$nameserver``` will be used in ```sed``` command to be appended to /etc/resolv.conf on SUT machine.

Reply:
So I think I can not use API to read all lines or bytes all together.

Reply:
I still assume there is a way to make this look more readable. Anyone from @os-autoinst/tools-team has an idea?

Reply:
One could of course use `for my $line (split("\n", path($resolv_file)->slurp)) { … }`. That's likely preferable. Note that your code lacks a `close` call (if you use `Mojo::File` you don't need to care about it).

Reply:
Done. Use one-liner.